SparkBuilderGeneratorPlugin/src/com/helospark/spark/builder/handlers/codegenerator/builderfieldcollector/SuperClassSetterFieldCollector.java

```diff
@@ -0,0 +1,92 @@
+package com.helospark.spark.builder.handlers.codegenerator.builderfieldcollector;
+
+import static com.helospark.spark.builder.preferences.PluginPreferenceList.INCLUDE_SETTER_FIELDS_FROM_SUPERCLASS;
+import static java.util.Collections.emptyList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.core.dom.BodyDeclaration;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.SingleVariableDeclaration;
+import org.eclipse.jdt.core.dom.TypeDeclaration;
+
+import com.helospark.spark.builder.handlers.codegenerator.component.helper.CamelCaseConverter;
+import com.helospark.spark.builder.handlers.codegenerator.component.helper.TypeDeclarationFromSuperclassExtractor;
+import com.helospark.spark.builder.handlers.codegenerator.domain.BuilderField;
+import com.helospark.spark.builder.handlers.codegenerator.domain.SuperSetterBasedBuilderField;
+import com.helospark.spark.builder.preferences.PreferencesManager;
+
+/**
+ * Collects fields with setters from superclass. 
+ * @author helospark
+ */
+public class SuperClassSetterFieldCollector implements FieldCollectorChainItem {
+    private static final String SETTER_METHOD_PREFIX = "set";
+
+    private PreferencesManager preferencesManager;
+    private TypeDeclarationFromSuperclassExtractor typeDeclarationFromSuperclassExtractor;
+    private CamelCaseConverter camelCaseConverter;
+
+    public SuperClassSetterFieldCollector(PreferencesManager preferencesManager, TypeDeclarationFromSuperclassExtractor typeDeclarationFromSuperclassExtractor,
+            CamelCaseConverter camelCaseConverter) {
+        this.preferencesManager = preferencesManager;
+        this.typeDeclarationFromSuperclassExtractor = typeDeclarationFromSuperclassExtractor;
+        this.camelCaseConverter = camelCaseConverter;
+    }
+
+    @Override
+    public List<? extends BuilderField> collectFields(TypeDeclaration typeDeclaration) {
+        if (preferencesManager.getPreferenceValue(INCLUDE_SETTER_FIELDS_FROM_SUPERCLASS)) {
+            return collectFieldsRecursively(typeDeclaration);
+        } else {
+            return emptyList();
+        }
+    }
+
+    private List<SuperSetterBasedBuilderField> collectFieldsRecursively(TypeDeclaration typeDeclaration) {
+        List<SuperSetterBasedBuilderField> result = new ArrayList<>();
+        Optional<TypeDeclaration> superClassType = typeDeclarationFromSuperclassExtractor.extractTypeDeclarationFromSuperClass(typeDeclaration);
+
+        superClassType.ifPresent(type -> result.addAll(findParametersWithSettersInType(type)));
```

Comment:
It looks like for me an "atomic" operation and I think we should handle those together. Please consider using just on ifPresent.

Reply:
Yes, also considered it during development.
I fixed it now.